OpenBSD cross-process thread signal delivery from emulators/wine, plus
an include of sig_openbsd.h (OpenBSD clang only warns on implicit
declarations).

kill_thread(): clear the fsync waiter table entries of the killed thread.

Index: server/thread.c
--- server/thread.c.orig
+++ server/thread.c
@@ -62,6 +62,9 @@
 #include "security.h"
 
 #include "fsync.h"
+#ifdef __OpenBSD__
+#include "sig_openbsd.h"
+#endif
 
 /* thread queues */
 
@@ -1657,6 +1660,9 @@ void kill_thread( struct thread *thread,
     }
     kill_console_processes( thread, 0 );
     abandon_mutexes( thread );
+#ifdef __OpenBSD__
+    if (do_fsync()) fsync_cleanup_thread_waiters( thread->unix_tid );
+#endif
     signal_sync( thread->sync );
     if (violent_death) send_thread_signal( thread, SIGQUIT );
     cleanup_thread( thread );
@@ -1814,6 +1820,10 @@ DECL_HANDLER(init_first_thread)
     current->unix_pid = process->unix_pid = req->unix_pid;
     current->unix_tid = req->unix_tid;
 
+#ifdef __OpenBSD__
+    init_process_signal_targets( process );
+#endif
+
     if (!process->parent_id)
         process->affinity = current->affinity = get_thread_affinity( current );
     else
@@ -2223,6 +2233,26 @@ DECL_HANDLER(queue_apc)
             }
         }
         break;
+#ifdef __OpenBSD__
+    case APC_COPY_MEMORY:
+        unsigned int access = apc->call.copy_memory.write ? PROCESS_VM_WRITE : PROCESS_VM_READ;
+        process = get_process_from_handle( req->handle, access);
+
+        if (process && process != current->process)
+        {
+            /* duplicate the shared section handle into the target process */
+            obj_handle_t handle = duplicate_handle( current->process, apc->call.copy_memory.handle,
+                                                    process, 0, 0, DUPLICATE_SAME_ACCESS);
+
+            if (handle) apc->call.copy_memory.handle = handle;
+            else
+            {
+                release_object( process );
+                process = NULL;
+            }
+        }
+        break;
+#endif
     default:
         set_error( STATUS_INVALID_PARAMETER );
         break;
